src/validation.cpp

```diff
@@ -5747,6 +5756,18 @@ bool AreEnforcedValuesDeployed()
     return fEnforcedValuesIsActive;
 }
 
+bool AreCoinbaseCheckAssetsDeployed()
+{
+    if (fCheckCoinbaseAssetsIsActive)
+        return true;
+
+    const ThresholdState thresholdState = VersionBitsTipState(GetParams().GetConsensus(), Consensus::DEPLOYMENT_COINBASE_ASSETS);
+    if (thresholdState == THRESHOLD_ACTIVE)
+        fCheckCoinbaseAssetsIsActive = true;
+
+    return fCheckCoinbaseAssetsIsActive;
+}
+
```

Comment:
Why not have a common function called for this,simmilar code is repeated for the last softfork aswell,what i mean is see this code

```C++
bool AreEnforcedValuesDeployed()
{
    if (fEnforcedValuesIsActive)
        return true;

    const ThresholdState thresholdState = VersionBitsTipState(GetParams().GetConsensus(), Consensus::DEPLOYMENT_ENFORCE_VALUE);
    if (thresholdState == THRESHOLD_ACTIVE || thresholdState == THRESHOLD_LOCKED_IN)
        fEnforcedValuesIsActive = true;

    return fEnforcedValuesIsActive;
}

bool AreCoinbaseCheckAssetsDeployed()
{
    if (fCheckCoinbaseAssetsIsActive)
        return true;

    const ThresholdState thresholdState = VersionBitsTipState(GetParams().GetConsensus(), Consensus::DEPLOYMENT_COINBASE_ASSETS);
    if (thresholdState == THRESHOLD_ACTIVE)
        fCheckCoinbaseAssetsIsActive = true;

    return fCheckCoinbaseAssetsIsActive;
}
```
This code can be shortened by having a common function for this instead of duplicating code for each softfork

Reply:
Good, Idea. I will make sure to do that in a different release. I don't want to add lot of non mandatory changes with this PR.